src/login/logind-dbus.c

```diff
@@ -2038,6 +2044,68 @@ static usec_t nologin_timeout_usec(usec_t elapse) {
         return LESS_BY(elapse, 5 * USEC_PER_MINUTE);
 }
 
+int manager_load_scheduled_shutdown(Manager *m) {
+        _cleanup_fclose_ FILE *f = NULL;
+        _cleanup_free_ char *usec = NULL,
+               *warn_wall = NULL,
+               *mode = NULL,
+               *wall_message = NULL,
+               *uid = NULL,
+               *tty = NULL;
+        _cleanup_free_ char *unescaped = NULL;
+        int r;
+
+        r = parse_env_file(f, SHUTDOWN_SCHEDULE_FILE,
+                        "USEC", &usec,
+                        "WARN_WALL", &warn_wall,
+                        "MODE", &mode,
+                        "WALL_MESSAGE", &wall_message,
+                        "UID", &uid,
+                        "TTY", &tty);
+        if (r == -ENOENT)
+                return 0;
+        if (r < 0) {
+                log_debug_errno(r, "Failed to parse " SHUTDOWN_SCHEDULE_FILE ": %m");
+                goto out;
+        }
+
+        r = deserialize_usec(strnull(usec), &m->scheduled_shutdown_timeout);
+        if (r < 0)
+                goto out; /* without proper time no point handling the rest */
+
+        HandleAction handle = handle_action_from_string(mode);
+        if (handle != _HANDLE_ACTION_INVALID)
+                m->scheduled_shutdown_type = manager_item_for_handle(handle);
+        else {
+                log_debug("Failed to parse scheduled shutdown type '%s', ignoring.", mode);
+                m->scheduled_shutdown_timeout = 0;
+                r = -EINVAL;
+                goto out;
+        }
+
+        /* remaining settings are not crucial, so ignore errors */
+
+        if (warn_wall && parse_boolean(warn_wall) >= 0)
+                m->enable_wall_messages = r;
+
+        if(wall_message && cunescape(wall_message, 0, &unescaped) >= 0)
+                free_and_strdup_warn(&m->wall_message, empty_to_null(unescaped));
+
+        if (uid)
+                parse_uid(uid, &m->scheduled_shutdown_uid);
+
+        free_and_strdup_warn(&m->scheduled_shutdown_tty, empty_to_null(tty));
+
+        r = manager_setup_shutdown_timers(m);
+        if (r >= 0)
+                (void) manager_setup_wall_message_timer(m);
+
+out:
+        if (r < 0)
+                (void) unlink(SHUTDOWN_SCHEDULE_FILE);
+        return r;
```

Comment:
we'd usually do this like this:

```c
        …
        return 0;

fail:
        (void) unlink(…);
        return r;
}
```

i.e. do the label only for the failure path, so that we don't have to check `r` again, and have one indentation less.